lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -531,12 +531,12 @@ def build_module(self, trans: GalaxyWebTransaction, payload=None):
         inputs = payload.get("inputs", {})
         trans.workflow_building_mode = workflow_building_modes.ENABLED
         module = module_factory.from_dict(trans, payload, from_tool_form=True)
-        if "tool_state" not in payload:
-            module_state: Dict[str, Any] = {}
-            errors: ParameterValidationErrorsT = {}
-            populate_state(trans, module.get_inputs(), inputs, module_state, errors=errors, check=True)
-            module.recover_state(module_state, from_tool_form=True)
-            module.check_and_update_state()
+
+        module_state: Dict[str, Any] = {}
```

Comment:
Why this change ? I think this might break cloning with a passed tool state ? See https://github.com/galaxyproject/galaxy/pull/6085/files

Reply:
It all seems to work fine in testing, and I see this fixes accessing module_state if the `tool_state` is in fact passed. I assume this hasn't caused problems because we stopped passing `tool_state` back to the build_module endpoint (all changes come through `payload.get("inputs", {})` ?) and now we only do this for inputs, so I think that's fine ? On the next occasion we should expand tests for this endpoint.